documents/DeveloperGuide/ShaderGeneration.md

```diff
@@ -0,0 +1,427 @@
+# Shader Generation
```

Comment:
This seems excellent, I'm really glad to see a developer document for shader generation!